### ISSUE TITLE
It doesn't work

### ISSUE THREAD

**Issue by Nourtirga** [derysudrajat/compass-qibla#1]
I installed the apk for testing but both the compass and the needle don't rotate, I tried to implement the code in my project and I got the same result!

**Comment by derysudrajat** [comment]
Hallo @Nourtirga, have you checked your internet and the location is turned on? and also make sure your device has a gyro sensor, to help your problem maybe you can put the log here.

**Comment by Nourtirga** [comment]
The location and internet were enabled but it doesn't seem to work on my android 12 device, I checked with other qibla apps because I had already some qibla apps installed on my device they work fine! the compass and the needle are fixed and don't rotate at all.

**Comment by derysudrajat** [comment]
maybe you can show me your code, let me check it

**Comment by Nourtirga** [comment]
I get the same result as your apk, It asks for location permission in the first launch, after that nothing happens!
PS: The GPS is enabled and there is internet connexion.
![Screenshot_20220628-163103_Permission controller](https://user-images.githubusercontent.com/15570805/176220133-53b3ba0e-a749-4d92-8734-35295c318c81.png)
![Screenshot_20220628-163110_compass-qibla](https://user-images.githubusercontent.com/15570805/176220156-0d58c787-9dbb-42ad-81e4-1a682f07445b.png)

**Comment by derysudrajat** [comment]
based on your screenshot the location is not shown, do you allow the permission on the first screenshot?

**Comment by derysudrajat** [comment]
it is probably the last location is null

**Comment by Nourtirga** [comment]
Yes, I allowed the permission after taking the first screenshot and then I took the second screenshot,  other qibla apps work fine!
